app/geo/gui/src/ts/twxplore/gui/geo/attributeStrategies/attributeStrategies.ts

```diff
@@ -1,156 +1,142 @@
 // In an OOP Language -
 
-import { FeatureAttributeName } from "../states/map/FeatureAttributeName";
+import {
+  FeatureAttributeName,
+  FeatureAttributeKey,
+} from "../states/map/FeatureAttributeName";
+import {FilterNames} from "../states/map/FilterNames";
 
 // TypeScript
-interface FeatureAttributeTypes {
-    isNumeric() : boolean
-    isString(): boolean
-    getFilterType(): string
-    ignore(): boolean
-    attributeName: FeatureAttributeName
-}
-
-
- export abstract class FeatureAttribute implements FeatureAttributeTypes{
-    public attributeName: FeatureAttributeName 
-    constructor(attributeName: FeatureAttributeName) {
-        this.attributeName = attributeName
-    }
-    static AttributeTypeOf(attributeName: FeatureAttributeName): FeatureAttribute {
-        switch(attributeName) {
-            case FeatureAttributeName.timestamp: {
-               return TimestampFeatureAttribute.getInstance(attributeName)
-            }
-            case FeatureAttributeName.frequency: {
-                return FrequencyFeatureAttribute.getInstance(attributeName)
-            }
-            case FeatureAttributeName.transmissionPower: {
-                return TransmissionPowerFeatureAttribute.getInstance(attributeName)
-            }
-            
-            default : {
-                return IgnoreFeatureAttribute.getInstance(attributeName)
-                
-            }
-         }
-    }
-
-    ignore(): boolean {
-        return false
-    }
-    getFilterType(){
-        return 'none'
-    }
-
-    abstract isNumeric() : boolean 
 
-    abstract isString() : boolean 
-    
+export abstract class FeatureAttribute {
+  constructor() {}
+
+  static valueOf(name: string): FeatureAttribute {
+    console.log(FeatureAttributeName[name as FeatureAttributeKey]);
+    switch (FeatureAttributeName[name as FeatureAttributeKey]) {
+      case FeatureAttributeName.timestamp: {
+        return TimestampFeatureAttribute.getInstance();
```

Comment:
I prefer `static get instance()` if the singleton is lazily instantiated (not the case here) or `readonly instance = new SomeClass()` on the class if it's eagerly instantiated (the case here). getInstance() is Java style.